nipype/interfaces/afni/preprocess.py

```diff
@@ -2820,7 +2828,48 @@ class Warp(AFNICommand):
     """
     _cmd = '3dWarp'
     input_spec = WarpInputSpec
-    output_spec = AFNICommandOutputSpec
+    output_spec = WarpOutputSpec
+
+    def _run_interface(self, runtime):
+        runtime = super(Warp, self)._run_interface(runtime)
+
+        if self.inputs.save_warp:
+            import numpy as np
+            if not self.inputs.out_file:
+                warp_file = fname_presuffix(self.inputs.in_file,
+                                            suffix='_warp_transform.mat',
+                                            use_ext=False)                        
+            else:
+                warp_file = fname_presuffix(self.inputs.out_file,
+                                            suffix='_transform.mat',
+                                            use_ext=False)
+            np.savetxt(warp_file, [runtime.stdout], fmt=str('%s'))
```

Comment:
Rather than calculate the filename twice:

```Python
if self.inputs.save_warp:
    import numpy as np
    warp_file = self._list_outputs()['warp_file']
    np.savetxt(warp_file, [runtime.stdout], fmt='%s')
```

(Not sure why you have `str('%s')`. If that's important, feel free to re-add.)

Reply:
>(Not sure why you have str('%s'). If that's important, feel free to re-add.)

When I don't explicit the format, I have an error
TypeError:  Mismatch between array dtype ('<U209') and format specifier ('%.18e')

So we need to keep it I think.

Reply:
Oh, I meant `str('%s')` vs `'%s'` (or is that also what you meant?). The `str()` seems like it should be a null-op, unless there's some weird Python2/3 hackery going on here.